runtime/src/main/java/org/corfudb/runtime/view/StreamsView.java

```diff
@@ -38,15 +37,21 @@
      */
     public static final String CHECKPOINT_SUFFIX = "_cp";
 
+    /**
+     * This list should have references to all opened streams. The ViewsGarbageCollector
+     * will use this list to clear trimmed addresses from streams. Since new streams
+     * can be opened during a runtime garbage collection cycle, the collection has
+     * to be thread-safe.
+     */
     @Getter
-    Multimap<UUID, IStreamView> streamCache = Multimaps.synchronizedMultimap(HashMultimap.create());
+    List<IStreamView> openedStreams = new CopyOnWriteArrayList<>();
```

Comment:
is this name change going to break any consuming classes ?

Reply:
This internal data structure isn't exposed to outside, thankfully.

Reply:
Will change it to private, we shouldn't expose this.

Reply:
changed to private